### PR TITLE
add com.innstereo.innstereo

### PR DESCRIPTION
**Pull request by tobias47n9e**
I still have issues with the desktop icon and there might be some other issues. I am at the point where I could use some feedback to further improve the manifest.

Repo: https://gitlab.com/innstereo/innstereo
Website: www.innstereo.com

### CODE REVIEW

File: org.gnome.innstereo.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
+    "app-id": "org.gnome.innstereo",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.28",
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Why not 3.30?

File: org.gnome.innstereo.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    "modules": [
+        {
+            "name": "gfortran",
+            "buildsystem": "simple",
+            "build-commands": [ "/usr/lib/sdk/gfortran-62/install.sh" ]
+        },
+        {
+            "name": "lapack",
+            "buildsystem": "cmake",
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Not needed.

File: org.gnome.innstereo.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            "config-opts": [
+                "-DCMAKE_INSTALL_PREFIX=/app",
+                "-DCMAKE_INSTALL_LIBDIR=lib",
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Not needed

File: org.gnome.innstereo.json [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            "name": "gfortran",
+            "buildsystem": "simple",
+            "build-commands": [ "/usr/lib/sdk/gfortran-62/install.sh" ]
+        },
+        {
+            "name": "lapack",
+            "buildsystem": "cmake",
+            "builddir": true,
+            "config-opts": [
+                "-DCMAKE_INSTALL_PREFIX=/app",
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Not needed

File: org.gnome.innstereo.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        },
+        "python3-setuptools_scm.json",
+        "python3-numpy.json",
+        "python3-matplotlib.json",
+        "python3-mplstereonet.json",
+        "python3-scipy.json",
+        {
+            "name": "innstereo",
+            "buildsystem": "meson",
+            "config-opts": [ "--libdir=lib" ],
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Not needed

File: org.gnome.innstereo.json [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            "builddir": true,
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Not needed.

File: org.gnome.innstereo.json [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+            "sources": [
+                {
+                    "type": "git",
+                    "branch": "feature/flatpak",
+                    "url": "file:///home/tobias/git/innstereo"
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: The url is wrong. Should be fixed & you need a specific commit as Flathub is meant for stable & reproducible builds.

File: org.gnome.innstereo.json [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                }
+            ],
+            "post-install": [
+                 "glib-compile-schemas /app/share/glib-2.0/schemas/"
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: Should be handled upstream instead using meson post install scripts

File: org.gnome.innstereo.json (hunk 1 above)
Comment (tobias47n9e) [reply org.gnome.innstereo.json]: Good point. Will upgrade

File: org.gnome.innstereo.json [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.gfortran-62"
+    ],
+    "build-options": {
+        "cflags": "-O2 -g",
```
Comment (bilelmoussaoui) [inline org.gnome.innstereo.json]: If you upgrade to 3.30 you can remove the cflags & cxxflags as they are set by default now in the FreeDesktop 18.08 (GNOME 3.30 is based on it)

File: org.gnome.innstereo.json (hunk 1 above)
Comment (tobias47n9e) [reply org.gnome.innstereo.json]: Might be blocked by: https://github.com/flathub/org.freedesktop.Sdk.Extension.gfortran-62/issues/3 ?

File: org.gnome.innstereo.json (hunk 2 above)
Comment (tobias47n9e) [reply org.gnome.innstereo.json]: When I remove that line I get this error:

> lapack should not be configured & built in the lapack source directory
> You must run cmake in a build directory.

File: org.gnome.innstereo.json (hunk 8 above)
Comment (tobias47n9e) [reply org.gnome.innstereo.json]: I tried doing it like the examples: https://gitlab.com/innstereo/innstereo/blob/master/build-aux/meson/postinstall.py#L19

Not sure why that line is not compiling the schemas.

File: org.gnome.innstereo.json [hunk 10]
```diff
@@ -0,0 +1,84 @@
+{
+    "app-id": "org.gnome.innstereo",
```
Comment (nedrichards) [inline org.gnome.innstereo.json]: this app-id doesn't look correct. this isn't an official GNOME project is it?

Comment (tobias47n9e) [reply org.gnome.innstereo.json]: No the project is not an official Gnome project. Should I use "com.innstereo.innstereo"?

Comment (x80486) [reply org.gnome.innstereo.json]: According to FreeDesktop's [Upstream Metadata](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic):

> The ID must follow a reverse-DNS scheme, consisting of `{tld}`.`{vendor}`.`{product}`, for example `org.kde.Gwenview` or `com.hugski.ColorHug2`.

Comment (tobias47n9e) [reply org.gnome.innstereo.json]: I moved to com.innstereo.innstereo namespace

File: org.gnome.innstereo.json (hunk 1 above)
Comment (nedrichards) [reply org.gnome.innstereo.json]: I don't think so - that just means that it's in the GNOME 3.30 SDK already without needing an extension (although it'll likely move back out again for a future release). note: I have not tested this, it's just the implication of the comments.

**Comment by nedrichards** [comment]
bot, build com.innstereo.innstereo

File: com.innstereo.innstereo.json [hunk 11]
```diff
@@ -0,0 +1,84 @@
+{
+    "app-id": "com.innstereo.innstereo",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.28",
```
Comment (barthalion) [inline com.innstereo.innstereo.json]: 3.30 is out since a while, have you tried it? It also includes gfortran if I'm not mistaken.

**Comment by barthalion** [comment]
JSON files for python modules are missing.

File: com.innstereo.innstereo.json (hunk 11 above)
Comment (tobias47n9e) [reply com.innstereo.innstereo.json]: I tried around but was not successful. Not sure how set the configs for the included gfortran.

**Comment by barthalion** [comment]
bot, build com.innstereo.innstereo

**Comment by barthalion** [comment]
CI is unhappy about appdata:

```

**Comment by tobias47n9e** [comment]
@barthalion I am unsure how to correctly fix that. Is there a guide for the desktop icon somewhere?

**Comment by barthalion** [comment]
bot, build com.innstereo.innstereo

**Comment by tobias47n9e** [comment]
@barthalion I tried it locally also with the newer numpy 1.15.2 version. Not sure why it suddenly freezes in the middle of compiling.

File: com.innstereo.innstereo.json [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules": [
+        {
+            "name": "lapack",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline com.innstereo.innstereo.json]:
```suggestion
            "buildsystem": "cmake-ninja",
```

**Comment by nedrichards** [comment]
bot, build com.innstereo.innstereo

**Comment by nedrichards** [comment]
bot, build com.innstereo.innstereo

**Comment by nedrichards** [comment]
Back to numpy explosion: https://flathub.org/builds/#/builders/2/builds/5795

File: com.innstereo.innstereo.json [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options": {
```
Comment (bilelmoussaoui) [inline com.innstereo.innstereo.json]: Not needed.

**Pull request closed without merging**